### PR TITLE
basic awesome books

### PR DESCRIPTION
**Pull request by Reggeditt**
### In JavaScript file:
     [ ] Created a collection that keeps a list of books.

     [ ] Created a function to add a new book to the collection, with title and author.
     [ ] Created a function to remove a book from the collection.
     [ ] Displays all books saved in the collection in the top part of the page.
#### when a user clicks the "Add" button:
         [ ] A new book is added to the collection.
         [ ] The new book is displayed on the page.

#### when a user clicks the "Remove" button:
        [ ] The correct book is removed from the collection.
        [ ] The correct book dissapears from the page.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  const addBook = (book) => {
+    collectionData.push(book);
+    window.localStorage.setItem('collectionArray', JSON.stringify(collectionData));
+  };
+
+  formEl.addEventListener('submit', (event) => {
+    event.preventDefault();
+    const title = titleInputEl.value;
+    const author = authorInputEl.value;
+    const removeIndex = collectionData.length;
```
Comment (levy002) [inline index.js]: - [ ] Kindly make sure to provide the unique `id`. At this moment we are having a conflict with ids please make sure to provide unique identification.

  _**HInt: `const removeIndex = Date.now()`**_

File: index.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    const book = { title, author, bookID: removeIndex };
+    addBook(book);
+    renderBooks(collectionData);
+    window.location.reload();
+  });
```
Comment (levy002) [inline index.js]: - [ ] Kindly make sure to not reload the page every time we add a new book. You can easily remove the `window.location.reload()`.

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @TaiwoEnoch   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.js (hunk 1 above)
Comment (Reggeditt) [reply index.js]: Can you please provide more context on this review? we are struggling to understand why and how to refactor our code with respect to your suggestion. The bookID's are all unique when we check them on our end.

File: index.js (hunk 2 above)
Comment (Reggeditt) [reply index.js]: Okay sir.

File: index.js (hunk 1 above)
Comment (Reggeditt) [reply index.js]: @levy002

File: index.js (hunk 2 above)
Comment (AlphaNtihinduka) [inline index.js]: - [ ] You did great in implementing the add book functionality 👏 . However as the previous reviewer suggested, make sure that the page is not reloaded when a book is added. Kindly consider removing `window.location.reload()`

File: index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+  const removeBook = (a) => {
+    collectionData = collectionData.filter((book) => book.bookID !== a);
+    renderBooks(collectionData);
+    window.localStorage.setItem('collectionArray', JSON.stringify(collectionData));
+  };
```
Comment (AlphaNtihinduka) [inline index.js]: - [ ] [OPTIONAL] The functionality of the removed book is working as intended 👏 . However, there are a few features you may need to add to make the project more professional. If you noticed, when the book is removed, the book_ids are not updated. Kindly ensure to update the id of the books once the book is deleted 👍

**Review by AlphaNtihinduka — CHANGES_REQUESTED** [review]
## Hi @Reggeditt 🙋‍♂️
Good job working on your first contact form. You have done the design to the best expectation and incredibly well. 👏  

However, there are some issues mostly on updating the book ids when a book is deleted
You still need to implement few on the project but you are almost there!!! :watch: 

TO HIGHLIGHT:
- Your project is professional ✅
- Professional README file :100: 
- Your PR is professional ✅
- Well-structured files ✅

# STATUS: Required Changes :recycle: 

close

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or slack channel if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK :crossed_fingers: :four_leaf_clover:

**Review by julie-ify — APPROVED** [review]
Hi @Reggeditt and @TaiwoEnoch,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi Team,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

Your Code Review Request is marked as invalid for these reasons:

- [ ] Please submit every project milestone with a new pull request and kindly never use the same pull request for multiple project milestones submission.
 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

**Comment by TaiwoEnoch** [comment]
Thank you @Omar-Muhamad

**Pull request merged**